### PR TITLE
Using the FIP job Controller to query managed jobs

### PR DESCRIPTION
**Pull request by nandajavarma**

### CODE REVIEW

File: deployments/kustomization/deployment.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ spec:
         runAsUser: 1001
       containers:
       - name: fip-job-status-api
-        image: registry.sighup.io/poc/fip-job-status-api:unstable
+        image: nandajavarma/fip-job-status-api:latest
```
Comment (angelbarrera92) [inline deployments/kustomization/deployment.yaml:25]: Be careful with this :)

File: internal/server/server.go [hunk 2]
```diff
@@ -55,46 +54,30 @@ func Setup(cfg *config.JobStatusAPIConfig) *gin.Engine {
...
-		}
+func parseResponse(response []resultsv1.CheckResult) []resultsv1.CheckResultSpec {
+	parsedResponse := make([]resultsv1.CheckResultSpec, len(response))
+	for index, result := range response {
+		parsedResponse[index] = result.Spec
 	}
 
-	return jap, nil
+	return parsedResponse
 }
```
Comment (angelbarrera92) [inline internal/server/server.go:81]: More simpler than before :) Looks good

File: pkg/checkresults/checkresults.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	// KfipBaseLabel use this label as base of other labels like check or group.
+	KfipBaseLabel = "kfip.sighup.io"
+
+	// AppBaseLabel use this label as base of other labels like name or component.
+	AppBaseLabel = "app.kubernetes.io"
+)
```
Comment (angelbarrera92) [inline pkg/checkresults/checkresults.go:22]: I've added these constants to the fip-commons library if i recall correctly

File: pkg/checkresults/checkresults_test.go [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) 2021 SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```
Comment (angelbarrera92) [inline pkg/checkresults/checkresults_test.go:3]: If we don't add test here, consider removing this  file

File: scripts/e2e/tests.sh [hunk 5]
```diff
@@ -8,10 +8,19 @@ info(){
   echo -e "${BATS_TEST_NUMBER}: ${BATS_TEST_DESCRIPTION}" >&3
 }
 
-@test "Requirements" {
+@test "Deploy the checkresults CRD" {
     info
     deploy(){
-        kubectl apply -f examples/job.yaml
+        kubectl apply -f scripts/e2e/v1beta1.crd.yaml
```
Comment (angelbarrera92) [inline scripts/e2e/tests.sh:14]: Consider using a URL instead of copy-pasting the crd definition here

File: scripts/e2e/v1beta1.crd.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2021 SIGHUP s.r.l All rights reserved.
```
Comment (angelbarrera92) [inline scripts/e2e/v1beta1.crd.yaml:1]: https://github.com/sighupio/fip-job-status-api/pull/11/files#r671184440

**Review by angelbarrera92 — CHANGES_REQUESTED** [review]
Minor comments. 

overall LGTM, but missing some clarifications

File: scripts/e2e/tests.sh (hunk 5 above)
Comment (nandajavarma) [reply scripts/e2e/tests.sh:14]: I will wait for the merge request to be merged for this

File: scripts/e2e/v1beta1.crd.yaml (hunk 6 above)
Comment (nandajavarma) [reply scripts/e2e/v1beta1.crd.yaml]: done

File: deployments/kustomization/rbac.yaml [hunk 7]
```diff
@@ -16,6 +16,9 @@ rules:
 - apiGroups: ["batch"]
```
Comment (angelbarrera92) [inline deployments/kustomization/rbac.yaml:16]: Question here... Does this API needs `jobs` and `pods` permissions with this new approach?

File: examples/rendered.yaml [hunk 8]
```diff
@@ -137,7 +143,7 @@ spec:
         - configMapRef:
             name: jsa-envs-k9gdmm2mb5
         image: registry.sighup.io/poc/fip-job-status-api:unstable
-        imagePullPolicy: IfNotPresent
+        imagePullPolicy: Always
```
Comment (angelbarrera92) [inline examples/rendered.yaml:146]: be careful with this value for `imagePullPolicy`.

File: deployments/kustomization/rbac.yaml (hunk 7 above)
Comment (nandajavarma) [reply deployments/kustomization/rbac.yaml:16]: ah you are right. It does not. I will change this!

Comment (nandajavarma) [reply deployments/kustomization/rbac.yaml]: done!

File: examples/rendered.yaml (hunk 8 above)
Comment (nandajavarma) [reply examples/rendered.yaml:146]: with the unstable image, personally it makes more sense to me to use `Always`. If we were using a stable release, I would definitely stick to `IfNotPresent`.

**Pull request merged**